packages/teleport/src/Teleport.tsx

```diff
@@ -109,4 +120,6 @@ export default hot(Teleport);
 export type Props = {
   ctx: TeleportContext;
   history: History;
+  renderPublicRoutes?(children?: JSX.Element[]): any;
+  renderPrivateRoutes?(CustomMain?: JSX.Element): JSX.Element;
```

Comment:
wouldn't this typing be same as `renderPublicRoutes`?

Reply:
No because it's just one parent element `<WaitingRoom>...</WaitingRoom>`

Reply:
Don't think so because `CustomMain` is a single parent component. Though the return type for `renderPublicRoutes` should be `JSX.Element[]`.